src/axom/quest/IntersectionShaper.hpp

```diff
@@ -1466,6 +1441,458 @@ class IntersectionShaper : public Shaper
   }
 
 private:
+  /*!
+   * \brief Filter the input mesh so it does not have any degenerate segments
+   *        (consecutive points that are the same) and the points are increasing
+   *        order on the Z (really X axis).
+   *
+   * \param m The input mesh.
+   * \return A new mint::Mesh that has been cleaned up.
+   */
+  SegmentMesh* filterMesh(const SegmentMesh* m) const
+  {
+    // Number of points in polyline
+    int pointcount = m->getNumberOfNodes();
+
+    // We'll be filtering the points in the mesh. Make a new mesh to contain
+    // those points.
+    SegmentMesh* newm = new SegmentMesh(m->getDimension(), mint::SEGMENT);
+    newm->reserveNodes(pointcount);
+
+    SLIC_INFO(axom::fmt::format(
+      "{:-^80}",
+      axom::fmt::format(
+        " Checking contour with {} points for degenerate segments ",
+        pointcount)));
+
+    enum
+    {
+      R = 1,
+      Z = 0
+    };
```

Comment:
```suggestion
    constexpr int R = 1;
    constexpr int Z = 0;
```

imo this is more understandable. I had to google anonymous enums...

Reply:
Changed it.